manifests/config.pp

```diff
@@ -10,9 +10,13 @@
 # [*purge*]
 #   Bool. If set will make puppet remove stale config files.
 #
+# [*pretty_print_json*]
+#   Add whitespace to config to make it more readable. Defaults to `false`.
+#
 class consul::config(
   $config_hash,
   $purge = true,
+  $pretty = false,
```

Comment:
I think it would be better if this is not parameterized but rather inherited from `init.pp` so that we can use the same value across all configs (services, checks and watches). We should avoid some configs being pretty while other not.

In other words, something like this:
```
consul_sorted_json($config_hash, $consul::pretty_config)
```